src/main/java/bio/terra/controller/RepositoryApiController.java

```diff
@@ -238,12 +271,18 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     }
 
     @Override
-    public ResponseEntity<List<DatasetSummaryModel>> enumerateDatasets(
-            @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
-            @RequestParam(value = "limit", required = false, defaultValue = "10") Integer limit) {
-        validiateOffsetAndLimit(offset, limit);
-        List<DatasetSummaryModel> datasetSummaryModels = datasetService.enumerateDatasets(offset, limit);
-        return new ResponseEntity<>(datasetSummaryModels, HttpStatus.OK);
+    public ResponseEntity<EnumerateDatasetModel> enumerateDatasets(
+            @Valid @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
+            @Valid @RequestParam(value = "limit", required = false, defaultValue = "10") Integer limit,
+            @Valid @RequestParam(value = "sort", required = false, defaultValue = "created_date") String sort,
+            @Valid @RequestParam(value = "direction", required = false, defaultValue = "asc") String direction,
+            @Valid @RequestParam(value = "filter", required = false) String filter) {
```

Comment:
does the @Valid do anything here?

Reply:
I don't think so, was following the pattern that was previously here.